spec/spec_helper.rb

```diff
@@ -13,7 +13,7 @@
 require 'rspec/matchers/fail_matchers'
 require 'simplecov'
 SimpleCov.start do
-  minimum_coverage 99.66
+  minimum_coverage 99.43
```

Comment:
I think it would be good to add a new spec to ensure the preconditions are only called once, I think you can call a spy/mock in 
https://github.com/toptal/granite/blob/d1fecea6dbb6fecff11f0891d09b681073a93095/spec/lib/granite/action/performing_spec.rb#L15 and then check is only called once in after perform

Reply:
Spec added :+1: